lib/src/screens/settings/feed_settings_screen.dart

```diff
@@ -0,0 +1,700 @@
+                search: nullIfEmpty(search),
+              );
+
+          // Check BuildContext
+          if (!mounted) return;
+
+          // Prevent duplicates
+          final currentItemIds =
+              _pagingController.itemList?.map((e) => e.id) ?? [];
+          final newItems = newPage.items
+              .where((e) => !currentItemIds.contains(e.id))
+              .toList();
+
+          _pagingController.appendPage(newItems, newPage.nextPage);
+          break;
+
+        case ExploreType.people:
+          // Lemmy cannot search with an empty query
+          if (context.read<AppController>().serverSoftware ==
+                  ServerSoftware.lemmy &&
+              search.isEmpty) {
+            _pagingController.appendLastPage([]);
+            return;
+          }
+
+          final newPage = await context.read<AppController>().api.users.list(
+            page: nullIfEmpty(pageKey),
+            filter: filter,
+            search: search,
+          );
+
+          // Check BuildContext
+          if (!mounted) return;
+
+          // Prevent duplicates
+          final currentItemIds =
+              _pagingController.itemList?.map((e) => e.id) ?? [];
+          final newItems = newPage.items
+              .where((e) => !currentItemIds.contains(e.id))
+              .toList();
+
+          _pagingController.appendPage(newItems, newPage.nextPage);
+          break;
+
+        case ExploreType.domains:
+          final newPage = await context.read<AppController>().api.domains.list(
+            page: nullIfEmpty(pageKey),
+            filter: filter,
+            search: nullIfEmpty(search),
+          );
+
+          // Check BuildContext
+          if (!mounted) return;
+
+          // Prevent duplicates
+          final currentItemIds =
+              _pagingController.itemList?.map((e) => e.id) ?? [];
+          final newItems = newPage.items
+              .where((e) => !currentItemIds.contains(e.id))
+              .toList();
+
+          _pagingController.appendPage(newItems, newPage.nextPage);
+          break;
+
+        case ExploreType.all:
+          final newPage = await context.read<AppController>().api.search.get(
+            page: nullIfEmpty(pageKey),
+            search: search,
+          );
+
+          if (!mounted) return;
+
+          _pagingController.appendPage(newPage.items, newPage.nextPage);
+          break;
+      }
+    } catch (error) {
+      _pagingController.error = error;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    const chipPadding = EdgeInsets.symmetric(vertical: 6, horizontal: 4);
+
+    final currentExploreSort = exploreSortSelection(context).getOption(sort);
+    final currentExploreFilter = exploreFilterSelection(
+      context,
+      type,
+    ).getOption(filter);
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(
+          '${l(context).explore} ${context.watch<AppController>().instanceHost}',
```

Comment:
I feel like this should be updated to reflect you're editing the feed, more specifically, adding inputs.